src/Protocol/ActivityPub/Receiver.php

```diff
@@ -2025,4 +2027,137 @@ private static function hasArrived(string $id): bool
 	{
 		return DBA::exists('arrived-activity', ['object-id' => $id]);
 	}
+
+	public static function processC2SActivity(array $activity, int $uid, array $application)
+	{
+		$ldactivity = JsonLD::compact($activity);
+		if (empty($ldactivity)) {
+			Logger::notice('Invalid activity', ['activity' => $activity, 'uid' => $uid]);
+			return;
+		}
+
+		$type = JsonLD::fetchElement($ldactivity, '@type');
+		if (!$type) {
+			Logger::notice('Empty type', ['activity' => $ldactivity, 'uid' => $uid]);
+			return;
+		}
+
+		$object_id   = JsonLD::fetchElement($ldactivity, 'as:object', '@id') ?? '';
+		$object_type = self::fetchObjectType($ldactivity, $object_id, $uid);
+		if (!$object_type && !$object_id) {
+			Logger::notice('Empty object type or id', ['activity' => $ldactivity, 'uid' => $uid]);
+			return;
+		}
+
+		Logger::debug('Processing activity', ['type' => $type, 'object_type' => $object_type, 'object_id' => $object_id, 'activity' => $ldactivity]);
+		self::routeC2SActivities($type, $object_type, $object_id, $uid, $application, $ldactivity);
+		throw new \Friendica\Network\HTTPException\AcceptedException();
```

Comment:
Is this really the only one doing it so?

I really dislike this "throw out" mechanism during the runtime. Normally it should return something "normal", so the caller can chose what to do with it.

I don't know, what @MrPetovan thinks about it, but for me:
Throwing = unwanted exits
return = wanted exists

And I know we don't follow this paradigm everywhere, like for redirects (which already caused some regressions too :) )

Reply:
I agree, it's been confusing for us having "positive" exceptions.